plans/network/pingpong.go

```diff
@@ -187,6 +206,96 @@ func pingpong(runenv *runtime.RunEnv, initCtx *run.InitContext) error {
 	return nil
 }
 
+// Exchanges own data network address with other instance in the test, and returns their data addres
+func exchangeListenAddrs(ctx context.Context, runenv *runtime.RunEnv, client sync.Client) (string, error) {
+
+	// Retrieve own data IP
+	tcpAddr, err := getSubnetAddr(runenv.TestSubnet)
+	if err != nil {
+		return "", err
+	}
+
+	mynode := &ListenAddrs{}
+	mine := map[string]struct{}{}
+
+	mynode.Addrs = append(mynode.Addrs, tcpAddr.String())
+	for _, l := range mynode.Addrs {
+		mine[l] = struct{}{}
+	}
+
+	_ = client.MustSignalAndWait(ctx, sync.State("listening"), runenv.TestInstanceCount)
+
+	allAddrs, err := shareAddresses(ctx, client, runenv, mynode)
+	if err != nil {
+		return "", err
+	}
+
+	otherAddrs := []string{}
+	// filter addresses to remove any own address
+	for _, addr := range allAddrs {
+		if _, ok := mine[addr]; ok {
+			continue
+		}
```

Comment:
See the comment below, if you have a single address, can you go for `if addr = tcpAddr`?